src/main/java/org/thethingsnetwork/java/app/lib/Client.java

```diff
@@ -299,5 +321,20 @@ public void send(String _devId, byte[] _payload, int _port) throws MqttException
         data.put("port", _port != 0 ? _port : 1);
         mqttClient.publish(appId + "/devices/" + _devId + "/down", data.toString().getBytes(), 0, false);
     }
+    
+    /**
+     * Send a downlink message using pre-registered encoder
```

Comment:
Great! The Node.js lib now also supports arrays, but we can do that in a later PR if you like: https://github.com/TheThingsNetwork/node-app-lib/pull/17